homeassistant/components/maxcube.py

```diff
@@ -0,0 +1,104 @@
+"""
+Platform for the MAX! Cube LAN Gateway.
+
+For more details about this component, please refer to the documentation
+https://home-assistant.io/components/maxcube/
+"""
+
+from socket import timeout
+import logging
+import time
+from threading import Lock
+
+from homeassistant.components.discovery import load_platform
+from homeassistant.const import CONF_HOST, CONF_PORT
+import homeassistant.helpers.config_validation as cv
+import voluptuous as vol
+
+REQUIREMENTS = ['maxcube-api==0.1.0']
+
+_LOGGER = logging.getLogger(__name__)
+DOMAIN = 'maxcube'
+
+CUBE_HANDLE = None
+
+DEFAULT_PORT = 62910
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Establish connection to MAX! Cube."""
+    from maxcube.connection import MaxCubeConnection
+    from maxcube.cube import MaxCube
+
+    # Read Config
+    host = config.get(DOMAIN).get(CONF_HOST)
+    port = config.get(DOMAIN).get(CONF_PORT)
+
+    # Global Cube Handle
+    global CUBE_HANDLE
```

Comment:
I got the inspiration from here: https://home-assistant.io/developers/component_generic_discovery/
I do see however, that your approach is much better --> changed.